invoices/kv_sql_migration_test.go

```diff
@@ -0,0 +1,124 @@
+package invoices_test
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/clock"
+	invpkg "github.com/lightningnetwork/lnd/invoices"
+	"github.com/lightningnetwork/lnd/sqldb"
+	"github.com/stretchr/testify/require"
+)
+
+// TestMigrationWithChannelDB tests the migration of invoices from a bolt backed
+// channel.db to a SQL database. Note that this test does not attempt to be a
+// complete migration test for all invoice types but rather is added as a tool
+// for developers and users to debug invoice migration issues with an actual
+// channel.db file.
+func TestMigrationWithChannelDB(t *testing.T) {
+	// First create a shared Postgres instance so we don't spawn a new
+	// docker container for each test.
+	pgFixture := sqldb.NewTestPgFixture(
+		t, sqldb.DefaultPostgresFixtureLifetime,
+	)
+	t.Cleanup(func() {
+		pgFixture.TearDown(t)
+	})
+
+	makeSQLDB := func(t *testing.T, sqlite bool) *invpkg.SQLStore {
+		var db *sqldb.BaseDB
+		if sqlite {
+			db = sqldb.NewTestSqliteDB(t).BaseDB
+		} else {
+			db = sqldb.NewTestPostgresDB(t, pgFixture).BaseDB
+		}
+
+		executor := sqldb.NewTransactionExecutor(
+			db, func(tx *sql.Tx) invpkg.SQLInvoiceQueries {
+				return db.WithTx(tx)
+			},
+		)
+
+		testClock := clock.NewTestClock(time.Unix(1, 0))
+
+		return invpkg.NewSQLStore(executor, testClock)
+	}
+
+	migrationTest := func(t *testing.T, store1 *channeldb.DB, sqlite bool) {
+		store2 := makeSQLDB(t, sqlite)
+		ctxb := context.Background()
+
+		const batchSize = 11
+		err := invpkg.MigrateInvoicesToSQL(
+			ctxb, store1.Backend, store1, store2, batchSize,
```

Comment:
call `store1` boltdb maybe ?

Reply:
Renamed to `kvStore` and `sqlStore`. PTAL.